widget/tag_preview.lua

```diff
@@ -0,0 +1,180 @@
+                }
+            end
+        end
+
+        local client_box = wibox.widget {
+            {
+                nil,
+                {
+                    nil,
+                    img_box,
+                    nil,
+                    expand = "outside",
+                    layout = wibox.layout.align.horizontal
+                },
+                nil,
+                expand = "outside",
+                widget = wibox.layout.align.vertical
+            },
+            forced_height = math.floor(c.height * scale),
+            forced_width = math.floor(c.width * scale),
+            bg = client_bg,
+            border_color = client_border_color,
+            border_width = client_border_width,
+            shape = helpers.shape.rrect(client_radius),
+            widget = wibox.container.background
+        }
+
+        client_box.point = {
+            x = math.floor(c.x * scale),
+            y = math.floor(c.y * scale)
+        }
+
+        --[[
+        local pos_layout = wibox.widget {
+            client_box,
+            forced_height = prev_screen_height,
+            forced_width = prev_screen_width,
+            layout = wibox.layout.manual
+        }
+        --]]
+
+        client_list:add(client_box)
+
+    end
+
+    tag_preview_box:setup{
+        {
+            {
+                {
+                    client_list,
+                    forced_height = prev_screen_height,
+                    forced_width = prev_screen_width,
+                    bg = widget_bg,
+                    widget = wibox.container.background
+                },
+                layout = wibox.layout.align.horizontal
+            },
+            layout = wibox.layout.align.vertical
+        },
+        bg = widget_bg,
+        border_width = widget_border_width,
+        border_color = widget_border_color,
+        shape = helpers.shape.rrect(screen_radius),
+        widget = wibox.container.background
+    }
+end
+
+local enable = function(opts)
+    local tag_preview_image = false
+    local widget_x = dpi(20)
+    local widget_y = dpi(20)
+    local screen_radius = beautiful.tag_preview_widget_border_radius or dpi(0)
+    local client_radius = beautiful.tag_preview_client_border_radius or dpi(0)
+    local client_opacity = beautiful.tag_preview_client_opacity or 0.5
+    local client_bg = beautiful.tag_preview_client_bg or "#000000"
+    local client_border_color = beautiful.tag_preview_client_border_color or
+                                    "#ffffff"
+    local client_border_width = beautiful.tag_preview_client_border_width or
+                                    dpi(3)
+    local widget_bg = beautiful.tag_preview_widget_bg or "#000000"
+    local widget_border_color = beautiful.tag_preview_widget_border_color or
+                                    "#ffffff"
+    local widget_border_width = beautiful.tag_preview_widget_border_width or
+                                    dpi(3)
+
```

Comment:
Maybe it would be better to include the appearance variables in `opts` instead of using `beautiful` variables. So that the user can use tag previews with different styles in different places.

Reply:
@Nooo37 I think that appearance stuff like colors should stay with beautiful variables. Most users would not use this widget multiple times, and the ones that do will probably be able to change the source.